packages/grpc-js/src/server-call.ts

```diff
@@ -490,21 +495,21 @@ export class Http2ServerCallStream<
     }
   }
 
-  sendError(error: ServiceError) {
+  sendError(error: ServerErrorResponse | ServerStatusResponse) {
     const status: StatusObject = {
       code: Status.UNKNOWN,
-      details: error.hasOwnProperty('message')
+      details: ('message' in error)
         ? error.message
         : 'Unknown Error',
-      metadata: error.hasOwnProperty('metadata')
+      metadata: ('metadata' in error && error.metadata !== undefined)
         ? error.metadata
         : new Metadata(),
     };
 
-    if (error.hasOwnProperty('code') && Number.isInteger(error.code)) {
+    if ('code' in error && util.isNumber(error.code) && Number.isInteger(error.code)) {
```

Comment:
The `util.isNumber()` (or `typeof error.code === 'number')` check shouldn't be necessary due to the `Number.isInteger()` call.

Reply:
I think Typescript was not happy with calling `Number.isInteger` on that value without verifying that it was a number.

Reply:
I think you're right: https://github.com/Microsoft/TypeScript/issues/21199 😕

Reply:
I think you're right: https://github.com/Microsoft/TypeScript/issues/21199 😕